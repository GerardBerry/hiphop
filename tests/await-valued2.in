;
!eval-begin;
var prg = example.prg;
var foo2 = example.foo2;
var foo3 = example.foo3;
prg.addEventListener("O", foo2);
!eval-end;
I(34);
!eval-begin;
prg.on("O", foo3)
!eval-end;
I;
!eval-begin;
prg.removeEventListener("O", foo3);
!eval-end;
I(15);
