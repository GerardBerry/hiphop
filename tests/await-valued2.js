"use hopscript"

var hh = require("hiphop");

var inSig = {accessibility: hh.IN};
var outSig = {accessibility: hh.OUT};

function foo(evt) {
   console.log("foo called by", evt.signalName, "with value", evt.signalValue);
}

function foo2(evt) {
   console.log("foo2 called by", evt.signalName, "with value", evt.signalValue);
}

function foo3(evt) {
   console.log("foo3 called by", evt.signalName, "with value", evt.signalValue);
}


var prg = <hh.module I=${inSig} O=${outSig}>
  <hh.loop>
    <hh.sequence>
      <hh.await I />
      <hh.emit O apply=${function() {return this.value.I}} />
    </hh.sequence>
  </hh.loop>
</hh.module>;

var m = new hh.ReactiveMachine(prg, "awaitvalued2");

m.addEventListener("O", foo);

console.log(";")
console.log(m.react());

m.addEventListener("O", foo2);

console.log("I(34)")
console.log(m.inputAndReact("I", 34));

m.addEventListener("O", foo3);

console.log("I");
console.log(m.inputAndReact("I"));

m.removeEventListener("O", foo3);

console.log("I(15)");
console.log(m.inputAndReact("I", 15));
