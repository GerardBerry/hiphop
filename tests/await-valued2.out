awaitvalued2> ;
--- Output:
awaitvalued2> I(34);
--- Output: O(34)
foo called by O with value 34
foo2 called by O with value 34
awaitvalued2> I;
--- Output: O(34)
foo called by O with value 34
foo2 called by O with value 34
foo3 called by O with value 34
awaitvalued2> I(15);
--- Output: O(15)
foo called by O with value 15
foo2 called by O with value 15
